src/status_im2/common/toasts/view.cljs

```diff
@@ -80,7 +80,7 @@
                                      (create-timer)))))]
            ;; create auto dismiss timer, clear timer when unmount or duration changed
            (rn/use-effect (fn [] (create-timer) clear-timer) [duration])
-           (rn/use-unmount on-dismissed)
+           (rn/use-unmount #(on-dismissed id))
```

Comment:
Why do we need to pass the `id` here and create the anonymous function? It's questionable because the subscription `[:toasts/toast-cursor id :on-dismissed]` already returns a function that has access to the `id` value and the `identity` function doesn't care about any argument in this case.

I'm just wondering why it wouldn't work using just `(rn/use-unmount on-dismissed)`.

Reply:
@yqrashawn ?

Reply:
It's just to handle situations where id and on-dismissed are both dynamically generated and hard or not elegant enough to pass the id to the on-dismissed fn. 
It might be a premature optimization.